c10/test/CMakeLists.txt

```
@@ -9,7 +9,11 @@ if(BUILD_TEST)
     if(NOT MSVC)
       target_compile_options(${test_name} PRIVATE -Wno-unused-variable)
     endif()
-    target_link_libraries(${test_name} c10 gmock gtest gtest_main)
+    if(BUILD_LIBTORCHLESS)
+      target_link_libraries(${test_name} ${C10_LIB} gmock gtest gtest_main)
+    else()
+      target_link_libraries(${test_name} c10 gmock gtest gtest_main)
+    endif()
```

I think it's fine to just depend on ${C10_LIB} 
```suggestion
      target_link_libraries(${test_name} ${C10_LIB} gmock gtest gtest_main)
```